src/basic/log.c

```diff
@@ -588,6 +592,27 @@ static int log_do_header(
         return 0;
 }
 
+static void log_do_context(struct iovec *iovec, size_t iovec_len, size_t *n) {
+        assert(iovec);
+        assert(n);
+
+        if (iovec_len <= 1)
+                return;
+
+        LIST_FOREACH(ll, c, _log_context) {
+                STRV_FOREACH(s, _log_context->fields) {
+                        if (*n >= iovec_len - 1)
+                                break;
+
+                        iovec[(*n)++] = IOVEC_MAKE_STRING(*s);
+                        iovec[(*n)++] = IOVEC_MAKE_STRING("\n");
+                }
+
+                if (*n >= iovec_len - 1)
+                        break;
```

Comment:
This repeated check is annoying. Why not just drop the second check and `return` from the first? We don't expect any items in this list with empty fields, so we don't need to optimize for that.